File: src/m/miscellaneous/structtoobj.py
def structtoobj(obj, S):
    """STRUCTTOOBJ - Convert struct to obj

    NOTE: The following semantics are not Pythonic, but attempt to recreate the 
    same function from MATLAB.
    """

    # Get object and structure fields
    structfields = S.__dict__.keys()
    objprops = vars(obj)

    # Recover object properties
    for structfield in structfields:
        if structfield in objprops:
            fieldvalue = getattr(S, structfield)
            setattr(obj, structfield, fieldvalue)

    return obj

File: src/m/miscellaneous/structtoobj.m
function obj=structtoobj(obj,S),
%STRUCTTOOBJ - convert a struct to an object, copying over matching fields
%
%   Usage:
%      obj=structtoobj(obj,S);

	%Get object and structure fields
	structfields=fieldnames(S);
	objprops    =properties(class(obj));

	%recover object properties
	for i=1:length(structfields)
		fieldname=structfields{i};
		if ismember(fieldname,objprops)
			fieldvalue=getfield(S,fieldname);
			obj=setfield(obj,fieldname,fieldvalue);
		end
	end
end
